### ISSUE TITLE
OpenWeatherMap to Weather Icon mapping breaks with Nerd Fonts

### ISSUE THREAD

**Issue by rustbuckett** [kamek-pf/ntfd#10]
Even with Weather Icons installed, listed in the the Polybar config and the font entry referenced in the config as label-font, the icons are all wrong with Nerd Fonts installed. Even using the test lines at the console return the wrong icons, though admittedly, I also have Nerd Fonts defined in my alacritty config. I've looked up the correct unicode values on Nerd Font site and found the mappings in OpenWeatherMap.hs, but I don't' understand the values listed as '\xxxxx' and how to set it to the Nerd Font values. 

It seems like the Polybar config should be working with the alternate font, but it isn't. Following, are the relevant lines from my Polybar config. Maybe I'm missing something here.

```
font-0 = DejaVu Sans Mono Nerd Font:style=Regular:pixelsize=16
font-1 = Material Icons:style=Regular:pixelsize=16
font-3 = Weather Icons:style=Regular:pixelsize=16

[module/weather]
type = custom/script
exec = busctl --user -j get-property io.ntfd /weather openweathermap.strings RenderedTemplate | jq -r .data
interval = 60
format = <label>
label = %output%
label-font = 3
```

**Comment by kamek-pf** [comment]
Try playing with the index. According to the [polybar docs](https://github.com/polybar/polybar/wiki/Fonts#fonts), `label-font` index is 1-based.

In your case `label-font = 3` would refer to `font-4`, which doesn't exist, so I assume it's defaulting to your terminal font.

**Comment by kamek-pf** [comment]
I'm also not sure how the `format` and `label` property come into play here, my own config looks like this:

```
[module/weather]
type = custom/script
label-font = 2
exec = busctl --user -j get-property io.ntfd /weather openweathermap.strings RenderedTemplate | jq -r .data
interval = 300
```

**Comment by rustbuckett** [comment]
> I'm also not sure how the `format` and `label` property come into play here, my own config looks like this:
> 
> ```
> [module/weather]
> type = custom/script
> label-font = 2
> exec = busctl --user -j get-property io.ntfd /weather openweathermap.strings RenderedTemplate | jq -r .data
> interval = 300
> ```

`format` and `label` aren't really necessary. They should default to the settings that I have in the config.

I tried changing the `label-font` to different numbers, and nothing resulted in correct icons.

Can you tell me about the icon mapping in OpenWeatherMap.hs? What is the \xxxxx number? Is it octal? How can I translate from that to a unicode number?

I ran into a similar problem with a shell script that I found that displays the weather and I had to update all of the mappings to Nerd Font icons.

**Comment by kamek-pf** [comment]
They are unicode code points converted from hex to base 10. In gucharmap, under character details the base 10 representation is shown as 'XML decimal entity'.

I missed it when I read your config the first time, but the cause of your issue is probably due to using a Nerd font as the very first font.
Try to use it as the last font (in your case, make it `font-3`). From the docs again:
> If you don't explicitly set the font-index for a tag, the bar will iterate the font list and pick the first one that is able to draw the character.
Also, it appears `label-font` actually does nothing for script modules, so what's probably happening here is that Polybar tries the first font and since Nerd has everything but under different code points, it renders the wrong thing and stops there.

Also, running busctl in your terminal will render the wrong characters for similar reasons, so it's not a good test, you need to actually run polybar to see the what happens.

**Comment by rustbuckett** [comment]
I tried changing the font ordering, and it wrecked all of the modules on the bar. I was able to explicitly declare a font for the module listed before nftd weather, but I was unable to effect any change on other modules. I don't know if that has something to do with the ordering of the modules since I didn't try rearranging them to further test the fonts. I found a simple bash script module that I was able to get working for my needs, but I still had to create a custom font/icon mapping in the script.
